Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -911,5 +918,32 @@ public static class Extensions {
             }
             return prettyPrinted.ToString().Trim();
         }
+
+        public static void RunSynchronouslyOnUIThread(this System.Threading.Tasks.Task task, IServiceProvider serviceProvider, [CallerMemberNameAttribute] string callerMemberName = "") {
+            if (!Dispatcher.CurrentDispatcher.CheckAccess()) {
+                string message = string.Format("{0} must be called on a UI thread.", callerMemberName);
+                Debug.Fail(message);
+                return;
```

Comment:
This is a replacement for `task.Result` rather than `await` - you'd use it where there's an async API, and you need to call it from someplace that has to be sync. In our case it's the data tip service, because it has to implement the VS interface for that stuff, which is sync.